platform/source/mbed_retarget.cpp

```diff
@@ -768,13 +881,31 @@ extern "C" int PREFIX(_read)(FILEHANDLE fh, unsigned char *buffer, unsigned int
 
 extern "C" ssize_t read(int fildes, void *buf, size_t length)
 {
+    if (length == 0 || buf == nullptr) {
```

Comment:
You should not be returning 0 for a null buffer. If you really want to test for null, `-1`+`EINVAL` would be an appropriate return.

But I maintain that you should not be wasting ROM space testing for null. We're writing in C/C++ here. Every single `memcpy`, pointer dereference or access to `this` could cause a fault, for approximately 99% of possible pointer values. Chucking in a few "if null" checks in places doesn't do anything, apart from give you something to unit test.

This sort of defensive logic would put `if (this == nullptr) return xxx;` on every single public method.

Reply:
I agree with you and was originally in favour of not adding it. However, @evedon has a different opinion I would therefore let her comment on this.

Reply:
First, I agree with the comment that if the buffer is null then we should not return 0.

Now regarding the check for null, I am used to defensive programming but indeed, having checked the mbed_retarget.cpp file, we are not checking for null buffer in other functions so it does not make sense to test in this function only.

So, let's remove the check for nullptr here.